QRCoder/PdfByteQRCode.cs

```diff
@@ -58,167 +51,205 @@ private byte[] HexColorToByteArray(string colorString)
-            "/XObject << /Im1 4 0 R >> >>\r\n" +
-            "/Contents 3 0 R\r\n" +
-            ">> ]\r\n" +
-            ">>\r\n" +
-            "endobj\r\n"
-        );
-
-        var X = "q\r\n" +
-            pdfMediaSize + " 0 0 " + pdfMediaSize + " 0 0 cm\r\n" +
-            "/Im1 Do\r\n" +
-            "Q";
-
-        writer.Flush();
-        xrefs.Add(stream.Position);
-
-        writer.Write(
-            xrefs.Count.ToString() + " 0 obj\r\n" +
-            "<< /Length " + X.Length.ToString() + " >>\r\n" +
-            "stream\r\n" +
-            X + "endstream\r\n" +
-            "endobj\r\n"
-        );
-
-        writer.Flush();
-        xrefs.Add(stream.Position);
-
-        writer.Write(
-            xrefs.Count.ToString() + " 0 obj\r\n" +
-            "<<\r\n" +
-            "/Name /Im1\r\n" +
-            "/Type /XObject\r\n" +
-            "/Subtype /Image\r\n" +
-            "/Width " + imgSize.ToString() + "/Height " + imgSize.ToString() + "/Length 5 0 R\r\n" +
-            "/Filter /DCTDecode\r\n" +
-            "/ColorSpace /DeviceRGB\r\n" +
-            "/BitsPerComponent 8\r\n" +
-            ">>\r\n" +
-            "stream\r\n"
-        );
-        writer.Flush();
-        stream.Write(jpgArray, 0, jpgArray.Length);
-        writer.Write(
-            "\r\n" +
-            "endstream\r\n" +
-            "endobj\r\n"
-        );
-
-        writer.Flush();
-        xrefs.Add(stream.Position);
-
-        writer.Write(
-            xrefs.Count.ToString() + " 0 obj\r\n" +
-            jpgArray.Length.ToString() + " endobj\r\n"
-        );
-
-        writer.Flush();
-        var startxref = stream.Position;
-
-        writer.Write(
-            "xref\r\n" +
-            "0 " + (xrefs.Count + 1).ToString() + "\r\n" +
-            "0000000000 65535 f\r\n"
-        );
-
-        foreach (var refValue in xrefs)
-            writer.Write(refValue.ToString("0000000000") + " 00000 n\r\n");
-
-        writer.Write(
-            "trailer\r\n" +
-            "<<\r\n" +
-            "/Size " + (xrefs.Count + 1).ToString() + "\r\n" +
-            "/Root 1 0 R\r\n" +
-            ">>\r\n" +
-            "startxref\r\n" +
-            startxref.ToString() + "\r\n" +
-            "%%EOF"
-        );
-
-        writer.Flush();
-
-        stream.Position = 0;
+        return pathCommands.ToString();
+    }
 
-        return stream.ToArray();
+    /// <summary>
+    /// Converts RGB byte array to PDF color space values (0.0 to 1.0).
+    /// </summary>
+    /// <param name="color">RGB color as byte array.</param>
+    /// <returns>PDF color string (three decimal values).</returns>
+    private static string ColorToPdfRgb(byte[] color)
+    {
+        if (color.Length != 3)
+            throw new ArgumentException("Color must be a 3-byte RGB array", nameof(color));
+
+        var r = ToStr(color[0] / (float)255);
+        var g = ToStr(color[1] / (float)255);
+        var b = ToStr(color[2] / (float)255);
```

Comment:
Just nice to have, for a tad better perf:
```suggestion
        const float inv255 = 1 / 255f;
        var r = ToStr(color[0] * inv255);
        var g = ToStr(color[1] * inv255);
        var b = ToStr(color[2] * inv255);
```

In strict term for floating point operations, that transform is not 100 % correct, but the error should be towards 0, so no real problem.